DataStructuresMemories/ViewControllers/DataStructViewController.swift

```diff
@@ -8,49 +8,103 @@
 
 import UIKit
 
-class DataStructViewController: UIViewController {
+class DetailViewController: UIViewController {
     
     @IBOutlet weak var textInsideLable: UILabel!
     
-    var tappedCell: DataStruct?
+    var tappedCell: DataStructProtocol?
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        self.navigationItem.title = tappedCell?.title
+        if let title = tappedCell?.title {
+            self.navigationItem.title = title
+        }
         textInsideLable?.text = tappedCell?.descr
         gradientView.opacityGradient()
     }
     
     @IBOutlet weak var gradientView: UIView!
     
+    var isTextShrinked = true {
+        didSet {
+            let title = isTextShrinked ? "More" : "Less"
+            let priorityForShrink = isTextShrinked ? UILayoutPriority(250) : UILayoutPriority(999)
+            let priorityForUnShrink = isTextShrinked ? UILayoutPriority(999) : UILayoutPriority(250)
+            UIView.transition(
+                with: childView,
+                duration: 0.3,
+                options: [.transitionFlipFromTop],
+                animations: { [ weak self ]  in
+                    self?.toggleButton.setTitleForAllStates(using: title)
```

Comment:
`guard let strongSelf = self else { return }` to make the code clear and avoid force unwrap